src/Native/configure.cmake

```diff
@@ -171,6 +171,36 @@ endif ()
 
 check_cxx_source_runs(
     "
+    #include <stdlib.h>
+    #include <time.h>
+    #include <sys/time.h>
+    int main()
+    {
+        int ret; 
+        struct timespec ts;
+        ret = clock_gettime(CLOCK_MONOTONIC, &ts);
+        exit(ret);
+    }
+    " 
+    HAVE_CLOCK_MONOTONIC)
+
+check_cxx_source_runs(
+    "
+    #include <stdlib.h>
+    #include <mach/mach_time.h>
+    int main()
+    {
+        int ret;
+        mach_timebase_info_data_t timebaseInfo;
+        ret = mach_timebase_info(&timebaseInfo);
+        mach_absolute_time();
+        exit(ret);
+    }
+    " 
+    HAVE_MACH_ABSOLUTE_TIME)
+
+check_cxx_source_runs(
+    "
```

Comment:
Can either of these just be check_function_exists or check_cxx_source_compiles? I used check_cxx_source_runs once by necessity for mmap issues, but I'm starting to wonder about the cross-compilation debt so it would be good to use it only where all else fails.